internal/v1/handler.go

```diff
@@ -27,6 +29,30 @@ const (
 	FSMaxSize = 68719476736
 )
 
+type links struct {
+	First string `json:"first"`
+	Last  string `json:"last"`
+}
+
+func (h *Handlers) newLinksWithExtraParams(path string, count, limit int, params url.Values) links {
+	lastOffset := count - 1
+	if lastOffset < 0 {
+		lastOffset = 0
+	}
+	fullPath := url.URL{Path: fmt.Sprintf("%v/v%v/%s", RoutePrefix(), h.server.spec.Info.Version, path)}
+
+	params.Add("offset", "0")
+	params.Add("limit", strconv.Itoa(limit))
+	fullPath.RawQuery = params.Encode()
+	first := fullPath.String()
+
+	params.Set("offset", strconv.Itoa(lastOffset))
+	fullPath.RawQuery = params.Encode()
+	last := fullPath.String()
+
+	return links{first, last}
+}
```

Comment:
Can you add a test for this one? I believe it is not correct (does not carry `offset` correctly for the 2nd page).

Reply:
There is a test for it... I'm testing it in the handler test :)

Reply:
Ok, my concern is that offset is not being added in some cases. Is that intentional?

Reply:
What do you mean? It's added to both first and last URL .... to one it's added (as it's first time its added to the params map and second time it's Set, as it's second time so we need to overriede the first value).
This is tested in the test...
I'm happy to back off with all these changes tho and just do a simple `fmt.Sprintf`

Reply:
Ah yeah now I see it, disregard.